src/main/scala/com/fulcrumgenomics/umi/FilterConsensusReads.scala

```diff
@@ -226,11 +246,15 @@ class FilterConsensusReads
   private[umi] def filterRecord(rec: SamRecord): FilterResult = {
     // Checks that are common to vanilla and duplex reads
     val maxDepth = rec.get[Int](ConsensusTags.PerRead.RawReadCount)
+    val minDepth = rec.get[Int](ConsensusTags.PerRead.MinRawReadCount)
     val errRate  = rec.get[Float](ConsensusTags.PerRead.RawReadErrorRate)
     if (maxDepth.isEmpty || errRate.isEmpty) fail(s"Read ${rec.name} does not appear to have consensus calling tags present.")
 
     // Only bother looking at reads where per-read criteria are met
-    if (maxDepth.exists(_ < this.ccFilters.minReads) || errRate.exists(_ > this.ccFilters.maxReadErrorRate)) {
+    if (maxDepth.exists(_ < this.ccFilters.minReads) ||
+      minDepth.exists(_ < this.ccFilters.minDepth) ||
+      errRate.exists(_ > this.ccFilters.maxReadErrorRate) ||
+      minMeanBaseQuality.exists(_ > rec.quals.map(_.toInt).sum.toDouble / rec.length)) {
```

Comment:
I think this would likely be more efficient as:

```scala
rec.quals.foldLeft(0)(_ + _).toDouble / rec.length
```

Since it'll avoid creation of an ArrayBuffer and Array[Int].

Reply:
Can you please add `this.` in front of `minMeanBaseQuality?  I see why you've done it, but it's confusing to me that the first three conditions are all `option from read </> filter value` and on the fourth one it's reversed.  Making it clear that the left hand side is an instance member would help.